src/components/CodeEditor/CodeEditor.jsx

```diff
@@ -26,24 +34,73 @@ export default function CodeEditor(props) {
     setChecked(event.target.checked);
   }
 
-  useEffect(() => {
-    checked ? setTheme("vs-dark") : setTheme("vs-light");
-  }, [checked]);
+  function checkError(data) {
+    let errorList = [];
+    let hasError = false;
+    data.errors.forEach((error, index) => {
+      if (data.errors[`${index}`].severity === "error" || data.errors[`${index}`].errorCode === "3420")
+      {
+        hasError = true;
+        var errorInfo = {
+          ERROR: `Compilation failed: ${data.errors[`${index}`].message} (type: ${data.errors[`${index}`].type}, code: ${data.errors[`${index}`].errorCode}).`
+        }
+        errorList.push(errorInfo);
+      }
+    })
+    if (hasError) {
+      return errorList;
+    }
+    return null;
```

Comment:
you could probably just return `errorList` , an empty array resolves to false when used in an if statement so you can return `[]` and then check for
```
if (errorList) {
  handleError...
}
```